pandas/io/parsers.py

```diff
@@ -1689,9 +1689,8 @@ def _convert_to_ndarrays(
                     values, set(col_na_values) | col_na_fvalues, try_num_bool=False
                 )
             else:
-                is_str_or_ea_dtype = is_string_dtype(
-                    cast_type
-                ) or is_extension_array_dtype(cast_type)
+                is_ea = is_extension_array_dtype(cast_type)
+                is_str_or_ea_dtype = is_string_dtype(cast_type) or is_ea
```

Comment:
put is_ea first to short-circuit the other check

Reply:
Done